daemon/cluster/cluster.go

```diff
@@ -429,6 +429,10 @@ func (c *Cluster) Init(req types.InitRequest) (string, error) {
 		return "", err
 	}
 
+	if net.ParseIP(advertiseHost).IsUnspecified() {
+		return "", errUnspecifiedAdvertiseAddr
```

Comment:
Need to call `c.Unlock()` here before returning - but moving the check inside `resolveAdvertiseAddr` will take care of that.